cltk/prosody/old_english/prosody.py

```diff
@@ -0,0 +1,115 @@
+from cltk.tag.pos import POSTag
+
+from cltk.corpus.old_english.alphabet import VOWELS
+from cltk.corpus.old_english.alphabet import LONG_VOWELS
+from cltk.corpus.old_english.alphabet import CONSONANTS
+
+tagger = POSTag('old_english')
+
+DIPHTHONGS = ['th', 'sc', 'ng', 'cg']
+
+
+def syllabify(w):
+    """
+    Syllabify OE word based on Onset Maximisation:
+
+    -Clusters of the form VCV are always partitioned as V-CV
+    - In case of VCCV clusters, the following are partitioned as V-CCV
+        - 'th' (considered the allophone of graphemes ⟨þ⟩ and ⟨ð⟩
+        - 'sc'
+        - 'ng'
+        - 'cg'
+
+    :param w: Word to be syllabified
+    :return: Syllabified word
+    """
+    syllables = []
+
+    for i, c in enumerate(w[1:-1]):
+        if c in CONSONANTS and w[i] in VOWELS and w[i+2] in VOWELS:
+            syllables.append(i)
+        if c in CONSONANTS and w[i] in VOWELS and w[i+2] in CONSONANTS:
+            if c + w[i+2] in DIPHTHONGS:
+                syllables.append(i)
+            else:
+                syllables.append(i+1)
+
+    for n, k in enumerate(syllables):
+        w = w[:k + n + 1] + "." + w[k + n + 1:]
+
+    w = w.split('.')
+
+    if sum([x in VOWELS for x in w[-1]]) == 0:
+        w[-2] = w[-2] + w[-1]
+        w = w[:-1]
+
+    return w
+
+def syllable_quantity(syl):
+    """
+    Return the syllable quantity:
+
+    Long syllables: CVC, CVV, CVVC
+    Short syllables: CV
+    :param syl:
+    :return:
+    """
+
+    if syl[-1] in VOWELS and syl[-2] in VOWELS: return 'LONG'
+    if syl[-1] in CONSONANTS and syl[-2] in VOWELS: return 'LONG'
+    if syl[-2] in VOWELS and syl[-1] in VOWELS: return 'LONG'
+    if syl[-1] in LONG_VOWELS: return 'LONG'
+
+    return 'SHORT'
+
+
+def stress_pos(text):
+    """
+    Rule-based stresser for Old English, taking into consideration the
+    POS as tagged by CLTK's trained perceptron tagger.
+
+    Note that the accuracy of the syllabifier is directly tied to that of
+    the tagger.
+
+    Primary stress rules in Old English are relatively simple:
+
+    - Compound words receive stress on the first component
+
+    - Verbal prefixes are never stressed
+
+    - 'ge-', 'be-' and 'for-' are never stressed
+
+    - Every other word receives primary stress on the first syllable
+    """
+
+    tagged_text = tagger.tag_crf(text)
+    f_text = []
+
+    for token, pos in tagged_text:
+
+        #Ignore conjuctions
+        if pos == 'C-':
+            continue
+
+        syl_token = syllabify(token.lower())
+        # if word is a verb, then receive stress on the second syllable
```

Comment:
Really? I thought that Germanic verbs (in the good old days) received stress at the first syllable, except for verbs with a specific prefix as ge-, be-, for-, etc. Maybe you should add at the top of your code sources which explain how stress can be assigned to Old English words.

Reply:
According to "Elementary Old English" (Sarelli, 2001) the stress is exactly as you say, @clemsciences:  stress on first syllable, except that verbs with prefixes receive stress on 2nd syllable.

Reply:
Oops, you're both right and I guess I should've clarified. There are still some pending changes one of them being the problem of prefixes . I think that the most straight-forward (and to me the only known) way to approach the problem, is to simply reference to a hard-coded prefix list (maybe as parts of alphabet.py), but I would really like to hear your opinion and whether a more elegant way exists.

Reply:
Yeah that's hard, because there are [so many prefixes](https://en.wiktionary.org/wiki/Category:Old_English_prefixes).  Even just `a-` is a prefix.  But short of training a classifier, requiring data, I'm not sure how to analyze the words other than by a dictionary.  

But would you need a complete dictionary, not just a list of prefixes?  That is, if the method is something like:

1,  Verb matches prefix P.
2. Verb without prefix  P matches verb V.
3.  => analyze as P + V, and give stress to 1st syllable of V.

But then one has to handle the inflections of V too, so lemmatization presumably is implied.  Unfortunately we don't have a lemmatizer for OE yet, right?

Would it be useful for me to work on a lemmatizer?  There's a couple of lexical sources we can consult.

Reply:
If you have enough data to work on a lemmatizer, go for it! It would tremendously help to develop a more abstracted approach instead of ad hoc rules.

Reply:
@Sedictious I'm looking into extracting lemmas and inflections from the English Wikitonary.  Coverage of Old English seems pretty good, though I don't have any stats.

I'm using gensim's https://radimrehurek.com/gensim/scripts/segment_wiki.html